src/api.ts

```diff
@@ -55,141 +89,93 @@ class ChatGPT {
-          case 404:
-            throw new SessionError('commands.chatgpt.messages.conversation-not-found')
-          case 429:
-            throw new SessionError('commands.chatgpt.messages.too-many-requests')
-          case 500:
-          case 503:
-            throw new SessionError('commands.chatgpt.messages.service-unavailable', [err.response.status])
-          default:
-            throw err
-        }
-      }
-    }
-
-    let response = ''
-    return await new Promise<Required<Conversation>>((resolve, reject) => {
-      let messageId: string
-      let conversationId: string
-      const parser = createParser((event) => {
-        if (event.type === 'event') {
-          const { data } = event
-          if (data === '[DONE]') {
-            return resolve({ message: response, messageId, conversationId })
+    const res = await this.page.evaluate((body, accessToken) => {
+      return new Promise(async (resolve: (value: string) => void, reject) => {
+        const decoder = new TextDecoder()
+        const res = await fetch('https://chat.openai.com/backend-api/conversation', {
+          method: 'POST',
+          body: body,
+          headers: {
+            authorization: `Bearer ${accessToken}`,
+            accept: 'text/event-stream',
+            'content-type': 'application/json',
           }
-          try {
-            const parsedData: types.ConversationResponseEvent = JSON.parse(data)
-            const message = parsedData.message
-            conversationId = parsedData.conversation_id
-
-            if (message) {
-              messageId = message?.id
-              let text = message?.content?.parts?.[0]
-
-              if (text) {
-                if (!this.config.markdown) {
-                  text = transform(text)
-                }
-
-                response = text
+        })
+
+        if (!res.ok) return reject(res.status)
+
+        let data: any
+        setTimeout(() => resolve(data), 2 * 60 * 1000) // There is browser environment, couldn't use `Time`
+        res.body.pipeTo(new WritableStream({
+          write(chunk) {
+            const chunks = decoder.decode(chunk).split('\n')
+            console.log('Receiving...')
+            for (const chunk of chunks) {
+              if (!chunk) continue
+              if (chunk.startsWith('data: [DONE]')) {
+                console.log('Done.')
+                return resolve(data)
               }
+              try {
+                const raw = chunk.replace('data: ', '')
+                JSON.parse(raw)
+                data = raw
+              } catch {}
             }
-          } catch (err) {
-            reject(err)
           }
-        }
+        }))
       })
-      data.pipe(new Writable({
-        write(chunk: string | Buffer, _encoding, cb) {
-          parser.feed(chunk.toString())
-          cb()
-        },
-      }))
-    })
+    }, JSON.stringify(body), accessToken)
+      .then(r => JSON.parse(r))
+      .catch((e: Error) => {
```

Comment:
Use `Quester.isAxiosError(e)` to narrow the type and then use `e.status` as the code rather than `.includes()`

Reply:
This error is rejected by browser and cannot be recognized by  `Quester.isAxiosError(e)`.